File: src/ELibrary.Backend/ShopApi/Features/StatisticsFeature/Domain/Models/BookStatistics.cs
namespace ShopApi.Features.StatisticsFeature.Domain.Models
{
    public class BookStatistics
    {
        public int InCartCopies { get; set; }
        public int InOrderCopies { get; set; }
        public int SoldCopies { get; set; }
        public int CanceledOrders { get; set; }
        public decimal AveragePrice { get; set; }
        public int StockAmount { get; set; }
        public decimal EarnedMoney { get; set; }
    }
}

File: src/ELibrary.Frontend/src/app/modules/shared/domain/models/shop/statistics/bookStatistics.ts

export interface BookStatistics {
    inCartCopies: number;
    inOrderCopies: number;
    soldCopies: number;
    canceledCopies: number;
    orderAmount: number;
    canceledOrderAmount: number;
    averagePrice: number;
    stockAmount: number;
    earnedMoney: number;
}

export function getDefaultBookStatistics(): BookStatistics {
    return {
        inCartCopies: 0,
        inOrderCopies: 0,
        soldCopies: 0,
        canceledCopies: 0,
        orderAmount: 0,
        canceledOrderAmount: 0,
        averagePrice: 0,
        stockAmount: 0,
        earnedMoney: 0,
    }
}